src/org/rascalmpl/library/lang/rascal/tests/basic/Locations.rsc

```diff
@@ -518,4 +518,11 @@ test bool extensionSetSimple()
 
 
 // we don't want backslashes in windows
-test bool correctTempPathResolverOnWindows() = /\\/ !:= resolveLocation(|tmp:///|).path;
\ No newline at end of file
+test bool correctTempPathResolverOnWindows() = /\\/ !:= resolveLocation(|tmp:///|).path;
+
+test bool encodingIsTheSameForPathAdditionAndTemplateInstantation()
```

Comment:
can we also make one that tests the behavior on windows?

Reply:
there is no OS dependent code in this part of the implementation. I.e. we do not use `new File` or anything like that or Path.separator. 

Perhaps we should?  Debatable. But right now we mimick exactly the behavior of the `+` semantics which also does not depend on OS specifics because it simply uses: `getValueFactory().sourceLocation(scheme, authority, newPath, query, fragment);`, via the fieldUpdate code on SourceLocationResult.

Reply:
```rascal
x = "c:\x\t.txt";
return |file:///<x>| == |file:///c:/x/t.txt|;
```

something like that should work on windows right?

Reply:
I'm pretty sure that it will not. There is no code in scope that will interpret the `\` as a path separator, so I'm pretty sure they will just stay where they are. 

Also, in this example you'd have to escape the backslashes: `x  = c:\\x\\t.txt`

Reply:
If we want OS-dependent behavior for path strings, I think this would be something for an actual RAP (Rascal Amendment Proposal). It will not be a backward compatible change.

Reply:
Yes, confirmed. the `+` operator and the fixed interpolator both will _not_ interpret windows path separators, even if on windows. It will just be `|file:///c:%5Cx%5Ct.txt|` where the `\` is escaped as `%5C` per the URI spec.

Reply:
This current fix solves the case where `x = "   "; |file:///<x>|;` would throw a URISyntaxException because there are spaces in the path. Instead it now behaves like `|file:///| + x` where the output would be `|file:///%20%20%20|`

The fix you have in mind requires a RAP. I'm open to the suggestion, but we'd have to carefully consider letting OS-dependent semantics into the Rascal language. The issue I have with that is that code that is run for a certain experiment could behave differently if run on another machine, with the same input data. So that's definitely worth another discussion.

Reply:
I'd rather put OS-dependent path conversion this under a well-documented library function than hide it under language semantics. Something like: `loc parseFilePath(str path, OS forOS=currentOS())`